proto/caraml/upi/v1/table.proto

```diff
@@ -0,0 +1,53 @@
+syntax = "proto3";
+
+package caraml.upi.v1;
+
+import "caraml/upi/v1/value.proto";
+
+
+// Table represents a 2D data structure that has one or more columns 
+// with potentially different types
+message Table {
+    // Table's name
+    string name = 1;
+    // Columns stores schema informations of all columns in the table.
+    repeated Column columns = 2;
+    // Rows stores list of row values in the table.
+    repeated Row rows = 3;
+}
+
+// Column represent a column definition within a table
+message Column {
+    // Column's name
+    string name = 1;
+    // Column's type
+    Type type = 2;
+}
+
+// Row represents list of value stored within a row of a table
+message Row {
+    // Row ID is defined by the client and can be used
+    // to join a prediction row with the prediction result,
+    // and to track predictions generated by multiple models.
+    // The user is expected include row ID (along with prediction ID)
+    // when calling the observations API so that predictions
+    // and observations can be joined. 
+    string row_id = 1;
+
+    // List of values within a row. 
+    // It is table's creator responsibility to ensure that the number of entry 
+    // values matches with the length of columns in the table.
```

Comment:
This feels dangerous. I assume the platform will validate len(values) == len(columns). but what about checking the types and order? is this the user's responsibility? there's potential for silent failures

Reply:
Yes, the table creator is also responsible to ensure the ordering of value matches the ordering of columns. I think the safety can be addressed separately by providing utility function to convert common data structure to UPI Table format. For example, in this PR I includes the example for python, we can add similar functionality to other language we supports.

Reply:
makes sense. some Go utilities might be useful for us

Reply:
We'll keep track this here https://github.com/caraml-dev/universal-prediction-interface/issues/8